src/api/server/authn.rst

```diff
@@ -345,3 +345,124 @@ headers to CouchDB with related requests:
 
 Note that you don't need to request :ref:`session <api/auth/session>`
 to be authenticated by this method if all required HTTP headers are provided.
+
+.. _api/auth/jwt:
+
+JWT Authentication
+====================
+
+.. note::
+    To use this authentication method make sure that the
+    ``{chttpd_auth, jwt_authentication_handler}`` value is added to the
+    list of the active :config:option:`chttpd/authentication_handlers`:
+
+    .. code-block:: ini
+
+        [chttpd]
+        authentication_handlers = {chttpd_auth, cookie_authentication_handler}, {chttpd_auth, jwt_authentication_handler}, {chttpd_auth, default_authentication_handler}
+
+`JWT authentication` enables CouchDB to use externally generated JWT tokens
+instead of defining users or roles in the ``_users`` database.
+
+The JWT authentication handler requires that all JWT tokens are signed by a key that
+CouchDB has been configured to trust (there is no support for JWT's "NONE" algorithm).
+
+Additionally, CouchDB can be configured to reject JWT tokens that are missing a
+configurable set of claims (e.g, a CouchDB administrator could insist on the ``exp`` claim).
+
+All claims presented in a JWT token are validated if presented, regardless of whether they
+are required.
+
+Two sections of config exist to configure JWT authentication;
+
+.. code-block:: ini
+
+    ;[jwt_auth]
+    ; List of claims to validate
+    ; required_claims =
+
+The `required_claims` config setting is a comma-separated list of additional mandatory
+JWT claims that must be present in any presented JWT token. A ``400 Bad Request`` is sent
+if any are missing.
+
+The ``alg`` claim is mandatory as it used to lookup the correct key for verifying the
+signature.
+
+The `sub` claim is mandatory and is used as the CouchDB user's name if the JWT token
+is valid.
+
+A private claim called `_couchdb.roles` is optional. If presented, it is a JSON
+array of strings, as is used as the CouchDB user's roles list if the JWT token
+is valid.
+
+.. code-block:: ini
+
+    ; [jwt_keys]
+    ; Configure at least one key here if using the JWT auth handler.
+    ; If your JWT tokens do not include a "kid" attribute, use "_default"
+    ; as the config key, otherwise use the kid as the config key.
+    ; Examples
+    ; hmac:_default = aGVsbG8=
+    ; hmac:foo = aGVsbG8=
+    ; The config values can represent symmetric and asymmetrics keys.
+    ; For symmetrics keys, the value is base64 encoded;
+    ; hmac:_default = aGVsbG8= # base64-encoded form of "hello"
+    ; For asymmetric keys, the value is the PEM encoding of the public
+    ; key with newlines replaced with the escape sequence \n.
+    ; rsa:foo = -----BEGIN PUBLIC KEY-----\nMIIBIjAN...IDAQAB\n-----END PUBLIC KEY-----\n
+    ; ec:bar = -----BEGIN PUBLIC KEY-----\nMHYwEAYHK...AzztRs\n-----END PUBLIC KEY-----\n
+
+The `jwt_key` section lists all the keys that this CouchDB server trusts. You
```

Comment:
Ideally this would link to the configuration portion of the documentation with the `:config:option` syntax.